[LimJH2002] iP

LimJH2002 opened the pull request:
# Simon Application Overview

> "Simplicity is the soul of efficiency." - [Austin Freeman](https://www.pinterest.com/pin/simplicity-is-the-soul-of-efficiency-austin-freeman-motivational-quote-of-the-day-august-15-2019-ave-mateiu--692850723914823746/#:~:text=Simplicity%20is%20the%20soul%20of,Austin%20Freeman%20Do%20you%20agree%3F)

**Simon** is an innovative application designed to assist users in managing their tasks. It's:
- text-based
- user-friendly
- ~~FAST~~ **Lightning-FAST** to use :zap:

**Getting Started**:
1. [Download Simon](https://github.com/LimJH2002/ip/releases) from our official repository.
2. Double-click the downloaded file to run.
3. Add your tasks using simple commands.
4. Let `Simon` manage everything for you! :smile:

**Key Features**:
- Task Management
- Event Tracking
- Deadline Monitoring

### For Java Programmers  

If you're a Java programmer, you can use Simon to practice your Java skills. Here's the main method to get you started:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Eclipse-Dominator committed:
Bump gradle and lib version

LimJH2002 committed:
Update Simon.java

LimJH2002 committed:
Added Echo functionality

LimJH2002 committed:
Added ability to add task and print task

LimJH2002 committed:
Added task class

LimJH2002 committed:
Added undone functionality

LimJH2002 committed:
Added mark functionality

LimJH2002 committed:
Fixed output bug

LimJH2002 committed:
Added SIMON Logo

LimJH2002 committed:
Added todo, event, deadlines classes

LimJH2002 committed:
Add support for tracking three types of tasks

LimJH2002 committed:
Update Simon.java

LimJH2002 committed:
Added testing

LimJH2002 committed:
Revamped codebase

LimJH2002 committed:
Added exception handling

LimJH2002 committed:
Added mark tasks error handling

LimJH2002 committed:
Added support for remove task

LimJH2002 committed:
Updated printing bug

LimJH2002 committed:
Fixed delete bug

LimJH2002 committed:
Updated test cases

LimJH2002 committed:
Added a lot more test cases

LimJH2002 committed:
Create simon.txt

LimJH2002 committed:
Fix typo in events

LimJH2002 committed:
Add saveTasksToFile function

LimJH2002 committed:
Add loadTasksFromFile

LimJH2002 committed:
Change field to protected

LimJH2002 committed:
Add checks for /data and simon.txt

LimJH2002 committed:
Update .gitignore

LimJH2002 committed:
Update .gitignore

LimJH2002 committed:
Update simon.txt

LimJH2002 committed:
Delete data directory

LimJH2002 committed:
Merge branch 'branch-Level-7'
Added Level 7 features

LimJH2002 committed:
Add date recognition for deadline class

LimJH2002 committed:
Fix deadline date bugs

LimJH2002 committed:
Update Simon.java

LimJH2002 committed:
Add support for datetime in events

LimJH2002 committed:
Add support to parse and record date time

LimJH2002 committed:
Added support for logic checkdate

LimJH2002 committed:
Add support for optional time in event

LimJH2002 committed:
Add support for optional time in deadline

LimJH2002 committed:
Merge branch 'branch-Level-8'
Completed Level-8 Features

LimJH2002 committed:
Add Ui Class

LimJH2002 committed:
Create Storage.java

LimJH2002 committed:
Create Parser.java

LimJH2002 committed:
Create TaskList.java

LimJH2002 committed:
Update TaskList

LimJH2002 committed:
Add more Ui support

LimJH2002 committed:
Update TaskList

LimJH2002 committed:
Update test

LimJH2002 committed:
Update Ui.java

LimJH2002 committed:
Update EXPECTED.TXT

LimJH2002 committed:
Merge branch 'branch-A-MoreOOP'
Complete OOP Structure of Simon the Chatbot

LimJH2002 committed:
Add package simon

LimJH2002 committed:
Complete stretch goal for package

LimJH2002 committed:
Merge branch 'branch-A-Packages'
Complete Package Tasks

LimJH2002 committed:
Merge branch 'add-gradle-support' of https://github.com/LimJH2002/ip into add-gradle-support

LimJH2002 committed:
Add support to run gradle

LimJH2002 committed:
Merge branch 'add-gradle-support'
Add Gradle support

LimJH2002 committed:
Add ParserTest

LimJH2002 committed:
Create TaskListTest.java

LimJH2002 committed:
Merge branch 'add-JUnit'
Add minimal JUnit test

LimJH2002 committed:
Add more test for Parser

LimJH2002 committed:
Fix parser bug

LimJH2002 committed:
Update Parser.java

LimJH2002 committed:
Update TaskListTest.java

LimJH2002 committed:
Update ParserTest.java

LimJH2002 committed:
Update TaskListTest.java

LimJH2002 committed:
Add StorageTest

LimJH2002 committed:
Create TaskTest.java

LimJH2002 committed:
Merge branch 'add-JUnit'
Add more testcases

LimJH2002 committed:
Update build.gradle

LimJH2002 committed:
Merge branch 'add-AJar'
Add Jar

LimJH2002 committed:
Add Javadoc for Parser

LimJH2002 committed:
Update Task.java

LimJH2002 committed:
Update Deadline.java

LimJH2002 committed:
Update Task.java

LimJH2002 committed:
Update ToDo.java

LimJH2002 committed:
Update Event.java

LimJH2002 committed:
Update Simon.java

LimJH2002 committed:
Update SimonAscii.java

LimJH2002 committed:
Update SimonException.java

LimJH2002 committed:
Update Storage.java

LimJH2002 committed:
Update TaskList.java

LimJH2002 committed:
Update Ui.java

LimJH2002 committed:
Update Parser.java

LimJH2002 committed:
Update Simon.java

LimJH2002 committed:
Update Storage.java

LimJH2002 committed:
Update Deadline.java

LimJH2002 committed:
Update Event.java

LimJH2002 committed:
Update TaskList.java

LimJH2002 committed:
Update Ui.java

LimJH2002 committed:
Update Parser.java

LimJH2002 committed:
Update Simon.java

LimJH2002 committed:
Update TaskList.java

LimJH2002 committed:
Update Ui.java

LimJH2002 committed:
Update TaskListTest.java

LimJH2002 committed:
Merge branch 'branch-A-JavaDoc'
Add JavaDoc

LimJH2002 committed:
Merge branch 'branch-A-CodingStandard'

LimJH2002 committed:
Merge branch 'branch-Level-9'

LimJH2002 committed:
Add checkstyle configuration

junnengsoo commented on src/main/java/simon/SimonAscii.java:
```diff
@@ -0,0 +1,26 @@
+public class SimonAscii {
+
+    /** ASCII representation of the name "Simon". */
+    private static final String name =
```
Really love the ASCII representation of SIMON!

junnengsoo commented on src/main/java/simon/Storage.java:
```diff
@@ -0,0 +1,131 @@
+package simon;
+
```
Love how you split the imports into their respective packages!

junnengsoo commented on src/test/java/simon/StorageTest.java:
```diff
@@ -0,0 +1,27 @@
+public class StorageTest {
+
+    @Test
+    public void load_validFile_tasksLoaded() throws SimonException {
```
Very nice code standards for test names!

junnengsoo reviewed (commented):
Amazing work!

LimJH2002 committed:
Update build.gradle

LimJH2002 committed:
Update build.gradle

LimJH2002 committed:
Run JavaFX properly

LimJH2002 committed:
Add dialog label

LimJH2002 committed:
Add scrolable vbox

LimJH2002 committed:
Create DialogBox.java

LimJH2002 committed:
Add avatar images

LimJH2002 committed:
Add iteration 2

LimJH2002 committed:
Add dialog box functionality

LimJH2002 committed:
Flip dialog box

LimJH2002 committed:
Update Main.java

LimJH2002 committed:
Add padding between each DialogBox

LimJH2002 committed:
Add padding between each ImageView and its Label

LimJH2002 committed:
Clip the ImageView into a circle

LimJH2002 committed:
Add fxml files

LimJH2002 committed:
Update MainWindow.fxml

LimJH2002 committed:
Move main code to Simon

LimJH2002 committed:
Fix all the bugs

LimJH2002 committed:
Update Main.java

LimJH2002 committed:
Integrate simon logic into the GUI

LimJH2002 committed:
Add styling into FXML

LimJH2002 committed:
Vertically center text

LimJH2002 committed:
Improve styling

LimJH2002 committed:
Remove text ui

LimJH2002 committed:
Merge branch 'branch-Level-10'
Complete level 10

LimJH2002 committed:
Add Vargs

neyapraveen commented on src/main/java/Simon.java:
```diff
@@ -1,12 +1,21 @@
     public static void main(String[] args) {
-        String output = "____________________________________________________________\n" +
+        String inData = "";
+        Scanner scan = new Scanner( System.in );
```
Perhaps could be improved by naming it inputScanner since scan could be interpreted as a verb.

neyapraveen commented on src/main/java/simon/Ui.java:
```diff
@@ -56,8 +55,7 @@ public void showError(String message) {
         clearOutput();
-        output.append("Got it. I've added this task:\n" + " " +
-                task + String.format("\nNow you have %d %s in the list.",
+        output.append("Got it. I've added this task:\n" + " ").append(task).append(String.format("\nNow you have %d %s in the list.",
```
Could line wrapping be improved since the first line is still rather long?

neyapraveen commented on src/main/java/simon/Ui.java:
```diff
@@ -69,7 +67,7 @@ public void showAddedTask(Task task, int count) {
     public void showDeletedTask(Task task, int count) {
         clearOutput();
-        output.append("Noted. I've removed this task:\n" + task + String.format("\nNow you have %d %s in the list.",
+        output.append("Noted. I've removed this task:\n").append(task).append(String.format("\nNow you have %d %s in the list.",
```
Could line wrapping be improved since the first line is still rather long?

neyapraveen commented on src/main/java/Simon.java:
```diff
@@ -209,7 +210,7 @@ private static void saveTasksToFile() {
                 } else if (task instanceof Deadline) {
                     Deadline deadline = (Deadline) task;
-                    writer.println("D | " + (task.isDone ? "1" : "0") + " | " + task.taskName + " | " + deadline.endDate);
+                    writer.println("D | " + (task.isDone ? "1" : "0") + " | " + task.taskName + " | " + deadline.endDateTime.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
```
Indentation for wrapped lines should be 8 spaces

neyapraveen commented on src/main/java/Deadline.java:
```diff
@@ -1,11 +1,23 @@
+
+public class Deadline extends Task {
+    protected final LocalDateTime endDateTime;
+    private static final DateTimeFormatter INPUT_FORMATTER = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```
Clear naming that explains the function

neyapraveen commented on src/main/java/simon/Simon.java:
```diff
@@ -0,0 +1,209 @@
+
+        try {
+            switch (command) {
+                case LIST:
```
switch statement should not have indentation for case clauses

neyapraveen commented on src/main/java/simon/Storage.java:
```diff
@@ -0,0 +1,131 @@
+     *
+     * @param tasks The {@code ArrayList} of tasks to be saved.
+     */
+    public void save(ArrayList<Task> tasks) {
```
Good naming, could be improved by naming function saveTaskToFile

neyapraveen reviewed (commented):
LGTM overall in terms of code quality! Some naming suggestions are given but overall code of high quality. Some nits to fix in terms of coding standard as well.

LimJH2002 committed:
Add assertions for Simon class

LimJH2002 committed:
Add asseertions to Tasklist

Add assertions to TaskList for better program correctness

Summary:
- Added assertions to various methods in the `TaskList` class to catch potential programming errors during development.

Details:
- Assert that tasks being added are not null.
- Ensure tasks being marked or deleted are not null.
- Validate that tasks retrieved from the list are not null.
- Check that the keyword used for searching tasks is not null or empty.

These assertions help to catch potential issues early in the development phase and ensure that the assumptions made in the code hold true throughout its execution.

Co-Authored-By: L1MJH <37625953+limjh2002@users.noreply.github.com>

LimJH2002 committed:
Merge pull request #1 from LimJH2002/branch-A-Assertions

Branch a assertions

LimJH2002 committed:
Update TaskList.java

LimJH2002 committed:
Improve code quality on Storage

LimJH2002 committed:
Improve code quality on Simon

LimJH2002 committed:
Update DialogBox.java

LimJH2002 committed:
Update Main.java

LimJH2002 committed:
Update Parser.java

LimJH2002 committed:
Update Event.java

LimJH2002 committed:
Merge pull request #2 from LimJH2002/branch-A-CodeQuality

Branch a code quality

LimJH2002 committed:
Add streams when reading from .txt

LimJH2002 committed:
Merge pull request #3 from LimJH2002/branch-streams

Add streams when reading from .txt

LimJH2002 committed:
Create gradle.yml

LimJH2002 committed:
Update SimonAscii.java

LimJH2002 committed:
Update ParserTest.java

LimJH2002 committed:
Update TaskListTest.java

LimJH2002 committed:
Update MainWindow.java

LimJH2002 committed:
Update Storage.java

LimJH2002 committed:
Update Deadline.java

LimJH2002 committed:
Update Deadline.java

LimJH2002 committed:
Update Event.java

LimJH2002 committed:
Update Parser.java

LimJH2002 committed:
Update ParserTest.java

LimJH2002 committed:
Update DialogBox.java

LimJH2002 committed:
Update TaskList.java

LimJH2002 committed:
Update Ui.java

LimJH2002 committed:
Update SimonAscii.java

LimJH2002 committed:
Update Parser.java

LimJH2002 committed:
Update Simon.java

LimJH2002 committed:
Update Event.java

LimJH2002 committed:
Update DialogBox.java

LimJH2002 committed:
Update MainWindow.java

LimJH2002 committed:
Update Simon.java

LimJH2002 committed:
Update Deadline.java

LimJH2002 committed:
Update Event.java

LimJH2002 committed:
Update Ui.java

LimJH2002 committed:
Update DialogBox.java

LimJH2002 committed:
Change assesor

LimJH2002 committed:
Update Event.java

LimJH2002 committed:
Update Task.java

LimJH2002 committed:
Update DialogBox.java

LimJH2002 committed:
Update DialogBox.java

LimJH2002 committed:
Update ParserTest.java

LimJH2002 committed:
Update StorageTest.java

LimJH2002 committed:
Update TaskTest.java

LimJH2002 committed:
Update TaskListTest.java

LimJH2002 committed:
Update gradle.yml

LimJH2002 committed:
Improve Parser code quality

LimJH2002 committed:
Update Parser.java

LimJH2002 committed:
Merge pull request #4 from LimJH2002/branch-CI

Branch ci

LimJH2002 committed:
Checks duplicate task

LimJH2002 committed:
More testing

LimJH2002 committed:
Update TaskListTest.java

LimJH2002 committed:
Update README.md

LimJH2002 committed:
Reduce method length

LimJH2002 committed:
Reduce load method length

LimJH2002 committed:
Edit style

LimJH2002 committed:
Better Gui

LimJH2002 committed:
Create Ui.png

LimJH2002 committed:
Update README.md

LimJH2002 committed:
Update README.md
